dashboard/app/models/concerns/localizable.rb

```diff
@@ -0,0 +1,172 @@
+#   course.localized_display_name('es', key_override: 'custom-key')  # Override lookup key
+#   course.localized_display_name_with_options(locale: 'es', key: 'custom-key')
+#
+#   # Internal usage - within the model itself
+#   class CourseOffering < ApplicationRecord
+#     include Localizable
+#     self.localizable_attributes = :display_name, :description
+#
+#     def formatted_title
+#       "Course: #{localized_display_name}"
+#     end
+#
+#    def summarize_for_show
+#       {
+#         key: key,
+#         display_name: localized_display_name,
+#         header: formatted_title,
+#         description: localized_description,
+#       }
+#     end
+#   end
+#
+# ## Features
+# - Automatic method generation for configured attributes
+# - Smart caching with cache invalidation
+# - Graceful fallback to original values
+# - Error handling for invalid locales
+# - Validation of configured attributes
+
+module Localizable
+  extend ActiveSupport::Concern
+
+  included do
+    after_initialize :setup_localization_cache
+  end
+
+  class_methods do
+    def localizable_attributes
+      @localizable_attributes || []
+    end
+
+    # Class method to define localizable attributes
+    def localizable_attributes=(*attrs)
+      @localizable_attributes = attrs.flatten.map(&:to_sym)
+
+      # Generate localized_* methods for each configured attribute
+      @localizable_attributes.each do |attr|
+        define_method("localized_#{attr}") do |locale_code = I18n.locale, key_override: nil|
+          localize_property(attr, locale_code: locale_code, key_override: key_override)
+        end
+
+        # Also create a method that accepts options hash (for backward compatibility)
+        define_method("localized_#{attr}_with_options") do |options = {}|
+          locale_code = options[:locale] || options[:locale_code] || I18n.locale
+          key_override = options[:key]
+          localize_property(attr, locale_code: locale_code, key_override: key_override)
+        end
+      end
+    end
+
+    # Helper to check if a class has any localizable attributes
+    def has_localizable_attributes?
+      localizable_attributes.any?
+    end
+  end
+
+  # Main localization method
+  def localize_property(property_name, locale_code: I18n.locale, key_override: nil)
+    property_name = property_name.to_sym
+    cache_key = "#{property_name}_#{locale_code}"
+
+    # Validate that the property is configured as localizable
+    unless self.class.localizable_attributes.include?(property_name)
+      CDO.log.warn("Attempting to localize non-configured attribute #{property_name} on #{self.class.name}")
+      return public_send(property_name) if respond_to?(property_name)
+      return nil
+    end
+
+    # Get the fallback value (original property value)
+    fallback_value = get_fallback_value(property_name)
+    # Return nil immediately if the fallback is nil (don't try to localize nil values)
+    if fallback_value.nil?
```

Comment:
Should we also skip empty strings?
```suggestion
    if fallback_value.blank?
```